scripts/sign.py

```diff
@@ -16,63 +16,156 @@ def int_parse(str):
 
 def get_args():
     from argparse import ArgumentParser
-
-    parser = ArgumentParser()
+    command_choices = ['sign', 'generate-digest', 'stitch-ta']
+
+    parser = ArgumentParser(
+        description='Sign a Tusted Application for OP-TEE.',
+        usage='\n   %(prog)s command [ arguments ]\n\n'
+
+        '   command:\n' +
+        '     sign             Generate signed loadable TA image file.\n' +
+        '                      Takes arguments --uuid, --in, --out' +
+        ' and --key.\n' +
+        '     generate-digest  Generate loadable TA binary image digest' +
+        ' for offline\n' +
+        '                      signing. Takes arguments  --uuid, --in and' +
+        ' --out.\n' +
+        '     stitch-ta        Generate loadable signed TA binary image' +
+        ' file from\n' +
+        '                      TA raw image and its signature. Takes' +
+        ' arguments\n' +
+        '                      --uuid, --in, --out, and --sig.\n\n' +
+
+        '   %(prog)s --help  show available commands and arguments\n\n',
+        epilog='If no command is given, the script will default to "sign".')
+
+    parser.add_argument(
+        'command', choices=command_choices, nargs='?',
+        default='sign',
+        help='Command, one of [' + ', '.join(command_choices) + ']')
     parser.add_argument('--uuid', required=True,
-                        type=uuid_parse, help='UUID of TA')
-    parser.add_argument('--version', type=int_parse, default=0, help='Version')
```

Comment:
https://github.com/OP-TEE/optee_os/pull/2832#discussion_r279693128

Reply:
OK, the reason I would keep it, is because it's not a lot of code and there is a default value. So we don't gain much by removing it.

You did not say why you think it has to go in the same patch as the rest.

Reply:
I assumed Etienne wanted it removed in this patch.

Aside from that common semantics for "--version" is printing the program Version, not passing signature Version number into the resulting binary without checking it in the code.

Reply:
OK, you convinced me. The version can be hardcoded since the script can evolve at the same time as the rest of the code.